pandas/tests/tools/test_to_datetime.py

```diff
@@ -1444,9 +1444,14 @@ def test_unit_mixed(self, cache, exp, arr):
         result = to_datetime(arr, errors="coerce", cache=cache)
         tm.assert_index_equal(result, expected)
 
-        msg = "mixed datetimes and integers in passed array"
-        with pytest.raises(ValueError, match=msg):
-            to_datetime(arr, errors="raise", cache=cache)
+        # GH#49037 pre-2.0 this raised, but it always worked with Series,
+        #  was never clear why it was disallowed
+        result = to_datetime(arr, errors="raise", cache=cache)
```

Comment:
Do we have a test where `unit` and/or `origin` is also specified in this mixed cases?

Reply:
i dont think so, no

Reply:
Could this test be easily added? ( I would expect `unit` and `origin` only applicable to the ints and ignored for datetimes )

Reply:
origin is only relevant with numeric dtypes, so won't go through array_to_datetime